frontend/views/pages/Payments.vue

```diff
@@ -213,6 +214,21 @@ export default ({
     this.setInitialActiveTab()
     this.updatePayments()
   },
+  mounted () {
+    const { section } = this.$route.query
+    const possibleSections = this.tabItems.map(tabItem => tabItem.url)
+    if (section && possibleSections.includes(section)) {
+      this.ephemeral.activeTab = section
+    } else {
+      const defaultTab = possibleSections[0]
+      if (defaultTab) {
+        this.handleTabClick(defaultTab)
+      } else if (section) {
+        const query = omit(this.$route.query, ['section'])
+        this.$router.push({ query }).catch(logExceptNavigationDuplicated)
+      }
+    }
+  },
```

Comment:
@Silver-IT it looks like the logic in this `mounted()` hook is more or less duplicate of what `$route` watcher below does. if the intention here is taking care of the same logic upon the first render of the component, I _think_ `immediate: true` flag in the watcher can be used instead. (reference in [PageSection.vue](https://github.com/okTurtles/group-income/blob/5f489c1fcc721d79ecebe688fbf27d49130396dd/frontend/views/components/PageSection.vue#L32C1-L41C6))

Reply:
Thank you, @SebinSong. I never knew that before.